### PR TITLE
1052: Refactors confirmation dialag and changes order on pageleave

### PR DESCRIPTION
**Pull request by DutchBen**
Refactored and rearranged a bit to make it clearer (to me at least :)) Let me know if I missewd something

### CODE REVIEW

**Review by Georgi2704 — APPROVED** [review]

**Review by Georgi2704 — APPROVED** [review]

File: packages/orchestrator-ui-components/src/contexts/ConfirmationDialogProvider.tsx [hunk 1]
```diff
@@ -87,11 +69,16 @@ export function ConfirmationDialogContextWrapper({ children }: any) {
...
-                cancel={(e) => state.confirmDialogAction(e, true)}
-                confirm={state.confirmDialogAction}
-                question={state.confirmationDialogQuestion}
-                leavePage={state.leavePage}
+                onCancel={() => setConfirmationDialogOpen(false)}
+                onConfirm={(e) => {
+                    state.onConfirm(e);
+                    setConfirmationDialogOpen(false);
+                }}
+                question={state.question}
```
[the remark this answers is not in the document]
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/contexts/ConfirmationDialogProvider.tsx:77]: done. I'm not convinced it makes it better though

File: packages/orchestrator-ui-components/src/contexts/ConfirmationDialogProvider.tsx [hunk 2]
```diff
@@ -1,83 +1,65 @@
...
-        leavePage,
-        isError,
-    }: ShowConfirmDialog) => {
+        isError = false,
+        subQuestion,
+        cancelButtonText,
+        confirmButtonText,
+    }: ConfirmDialogState) => {
         setConfirmationDialogOpen(true);
         setState({
```
[the remark this answers is not in the document]
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/contexts/ConfirmationDialogProvider.tsx:56]: Al values are set at the same time so I don't think it's needed

File: packages/orchestrator-ui-components/src/components/WfoForms/UserInputForm.tsx [hunk 3]
```diff
@@ -33,7 +33,8 @@ import {
     EuiHorizontalRule,
 } from '@elastic/eui';
 
-import { ConfirmDialogActions, ConfirmationDialogContext } from '@/contexts';
+import { ConfirmationDialogContext } from '@/contexts';
+import { ConfirmDialogHandler } from '@/contexts';
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WfoForms/UserInputForm.tsx:37]: Can be combined as a single line import.

Comment (DutchBen) [reply packages/orchestrator-ui-components/src/components/WfoForms/UserInputForm.tsx]: done

**Pull request merged**